entrypoint.sh

```diff
@@ -45,15 +46,54 @@ if [[ ! $RC ]]; then
 	exit 1
 fi
 
+echo "Setting ims job status to building_image"
 python3 -m ims_python_helper image set_job_status $IMS_JOB_ID building_image
 
 DEBUG_FLAGS=""
 if [[ `echo $ENABLE_DEBUG | tr [:upper:] [:lower:]` = "true" ]]; then
     DEBUG_FLAGS="--debug"
 fi
 
+# If this is running an arm64 emulation we need to set up the emulator, then
+# run the arm64 version of the ims-kiwi-build image.
+echo "Checking build platform: $BUILD_PLATFORM"
+if [ "$BUILD_PLATFORM" == "aarch64" ]; then
+    echo "Build platform is aarch64"
+    # Regiser qemu-aarch64-static to act as an arm interpreter for arm builds 
+    if [ ! -d /proc/sys/fs/binfmt_misc ] ; then
+        echo "- binfmt_misc does not appear to be loaded or isn't built in."
+        echo "  Trying to load it..."
+        if ! modprobe binfmt_misc ; then
+            echo "FATAL: Unable to load binfmt_misc"
+            exit 1;
+        fi
+    fi
+
+    # mount the emulation filesystem
+    if [ ! -f /proc/sys/fs/binfmt_misc/register ] ; then
+        echo "- The binfmt_misc filesystem does not appear to be mounted."
+        echo "  Trying to mount it..."
+        if ! mount binfmt_misc -t binfmt_misc /proc/sys/fs/binfmt_misc ; then
+            echo "FATAL:  Unable to mount binfmt_misc filesystem."
+            exit 1
+        fi
+    fi
+
+    # register qemu for aarch64 images 
+    if [ ! -f /proc/sys/fs/binfmt_misc/qemu-aarch64 ] ; then
+        echo "- Setting up QEMU for ARM64"
+        echo ":qemu-aarch64:M::\x7f\x45\x4c\x46\x02\x01\x01\x00\x00\x00\x00\x00\x00\x00\x00\x00\x02\x00\xb7\x00:\xff\xff\xff\xff\xff\xff\xff\x00\xff\xff\xff\xff\xff\xff\xff\xff\xfe\xff\xff\xff:/usr/bin/qemu-aarch64-static:F" >> /proc/sys/fs/binfmt_misc/register
```

Comment:
Well, that's a pretty ugly echo.

Reply:
Yes it is it, but that is how the binfmt_misc stuff is done. Needs to have the unique digital signature of an arm64 binary registered for it to work. We will clean up when we have some breathing room. :)